packages/cypress/src/migrations/update-17-2-0/add-nx-cypress-plugin.ts

```diff
@@ -0,0 +1,24 @@
+import { formatFiles, getProjects, Tree } from '@nx/devkit';
+import { createNodes } from '../../plugins/plugin';
+
+import { createProjectRootMappingsFromProjectConfigurations } from 'nx/src/project-graph/utils/find-project-for-path';
+import { replaceProjectConfigurationsWithPlugin } from '@nx/devkit/src/utils/replace-project-configuration-with-plugin';
+
+export default async function update(tree: Tree) {
+  const proj = Object.fromEntries(getProjects(tree).entries());
+
+  const rootMappings = createProjectRootMappingsFromProjectConfigurations(proj);
+
+  replaceProjectConfigurationsWithPlugin(
+    tree,
+    rootMappings,
+    '@nx/cypress/plugin',
+    createNodes,
+    {
+      targetName: 'e2e',
```

Comment:
I don't think this is safe, because someone may have named their target `katerina-e2e`. I think here we should use the `forEachExecutorOptions` to get the target names?

Reply:
so, if a project has a target defined as such
```
   "e2e-katerina": {
      "executor": "@nx/cypress:cypress",
      "options": {
...
      },
      "configurations": {
  ...
      }
    },
```

our migrator would create a new `e2e` node correctly, but it would not remove the `e2e-katerina` from `project.json`.

Reply:
there's an issue though if one has multiple cypress projects, each one with different target names. for a migration to work as intended, all target names of all projects should be the same 🤔

Reply:
I don't think this makes sense either, I'll suggest a way to handle this in new PR, and see if we care for such complexity.

Reply:
maybe I'm just overthinking this

Reply:
Let's land this as is for now and discuss this more. We might be able to figure out the target name but let's not block this change.